javascript/ql/lib/semmle/javascript/dataflow/Nodes.qll

```diff
@@ -960,7 +962,16 @@ class ClassNode extends DataFlow::SourceNode instanceof ClassNode::Range {
       result.getAstNode().getFile() = this.getAstNode().getFile()
     )
     or
-    exists(DataFlow::TypeTracker t2 | result = this.getAClassReference(t2).track(t2, t))
+    result = getAClassReferenceStep(t)
+  }
+
+  pragma[noopt]
+  private DataFlow::SourceNode getAClassReferenceStep(DataFlow::TypeTracker t) {
```

Comment:
```suggestion
  private DataFlow::SourceNode getAClassReferenceRec(DataFlow::TypeTracker t) {
```
We have other cases of a type-tracking "step" predicates being factored out to fix a join ordering, but where some information about the edge itself is matched. This here is just the whole recursive case being factored, so I'd prefer a different name. I was certainly confused for a second, before realising that the `noopt` is reason for factoring it out.